spec/support/vendor_api/vendor_api_spec_helpers.rb

```diff
@@ -21,8 +21,12 @@ def auth_headers
     { headers: { 'Authorization' => "Bearer #{unhashed_token}" } }
   end
 
-  def currently_authenticated_provider
-    @currently_authenticated_provider ||= create(:provider)
+  def currently_authenticated_provider(code: 'UCL')
+    @currently_authenticated_provider ||= create(:provider, code: code)
+  end
+
+  def alternate_provider(code: 'STR')
```

Comment:
What would you think of moving this into the test, to avoid the helpers to become too large?

Reply:
moved